pkg/platform/kube/functionres/lazy.go

```diff
@@ -597,6 +614,22 @@ func (lc *lazyClient) createOrUpdateDeployment(functionLabels labels.Set,
 	return resource.(*apps_v1beta1.Deployment), err
 }
 
+func enrichDeploymentSpec(dest *apps_v1beta1.DeploymentSpec, source apps_v1beta1.DeploymentSpec) error {
+	encodedDeploymentSpec, err := json.Marshal(dest)
+	if err != nil {
+		return errors.Wrap(err, "Failed to marshal destination deployment spec")
+	}
+
+	err = json.Unmarshal(encodedDeploymentSpec, &source)
+	if err != nil {
+		return errors.Wrap(err, "Failed to join source deployment with target deployment")
```

Comment:
Source/destination are a bit unclear as error messages here.
For error messages - we need to use statements that are understandable outside the scope of this method, e.g. `Failed to enrich function deployment configuration with platform deployment configuration`

Reply:
Also as local variable names they are not clear as to what is overriding what